backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClient.java

```diff
@@ -98,27 +100,19 @@ public void setReturnValue(UUID uuid) {
         returnValue.setLogFile(runnerLogger.getLogFile());
     }
 
-    public List<String> getSortedEvents(String playUuid, int lastEventId) throws InterruptedException {
-        Boolean artifactsIsPopulated = false;
-        List<String> sortedEvents = new ArrayList<>();
-
-        while (!artifactsIsPopulated) {
-            Thread.sleep(1500);
-
-            // ignoring incompleted json files, add to list only events that haven't been handles yet.
-            String jobEvents = getJobEventsDir(playUuid);
-            if (Files.exists(Paths.get(jobEvents))) {
-                sortedEvents = Stream.of(new File(jobEvents).listFiles())
-                        .map(File::getName)
-                        .filter(item -> !item.contains("partial"))
-                        .filter(item -> !item.endsWith(".tmp"))
-                        .filter(item -> (Integer.valueOf(item.split("-")[0])) > lastEventId)
-                        .sorted()
-                        .collect(Collectors.toList());
-                artifactsIsPopulated = true;
-            }
+    public String getNextEvent(String playUuid, int lastEventId) {
+        Optional<String> nextEvent = Optional.empty();
+        // ignoring incompleted json files, add to list only events that haven't been handles yet.
+        String jobEvents = getJobEventsDir(playUuid);
+        if (Files.exists(Paths.get(jobEvents))) {
+            nextEvent = Stream.of(new File(jobEvents).listFiles())
+                    .map(File::getName)
+                    .filter(item -> !item.contains("partial"))
+                    .filter(item -> !item.endsWith(".tmp"))
+                    .filter(item -> item.startsWith((lastEventId + 1) + "-"))
+                    .findFirst();
         }
-        return sortedEvents;
+        return nextEvent.isPresent() ? nextEvent.get() : null;
```

Comment:
for your consideration: I'd replace the use of Optional with early return (with null) in case the path doesn't exist and then return with findFirst().orElse(null) - a bit cleaner in my opinion